src/main/kotlin/domain/Board.kt

```diff
@@ -18,6 +19,40 @@ class Board(
 
     fun getNearByMineCount(height: Int, width: Int): Int {
         val coordinate = Coordinate(height, width)
-        return fields.getNearByFields(coordinate).filterIsInstance<Mine>().count()
+        return fields.getNearByFields(coordinate).count { it.field is Mine }
+    }
+
+    fun isMine(height: Int, width: Int): Boolean {
+        val coordinate = Coordinate(height, width)
+        return fields.getField(coordinate) is Mine
+    }
+
+    fun isGameOver(): Boolean {
+        return fields.isLandAllOpened()
+    }
+
+    tailrec fun open(height: Int, width: Int) {
+        // 해당 좌표 열기
+        val coordinate = Coordinate(height, width)
+        fields.open(coordinate)
+
+        // 인근에 지뢰가 아닌 좌표 열기
+        getNotOpenedLandAndHasNoMineWithCoordinate(coordinate).forEach { land ->
+            open(land.coordinate.row, land.coordinate.col)
+        }
+    }
+
+    private fun getNotOpenedLandAndHasNoMineWithCoordinate(coordinate: Coordinate): List<FieldWithCoordinate> {
+        if (hasMineNearby(coordinate))
+            return emptyList()
+        return getNotOpenedLand(coordinate)
+    }
+
+    private fun hasMineNearby(coordinate: Coordinate): Boolean {
+        return fields.getNearByFields(coordinate).count { it.field is Mine } > 0
```

Comment:
```suggestion
        return fields.getNearByFields(coordinate).any { it.field is Mine }
```
조금 더 간단하게 표현할 수 있겠네요.